wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/devfile/convert/component/editor/EditorComponentToWorkspaceApplier.java

```diff
@@ -92,6 +92,11 @@ public void apply(
 
     // make sure id is set to be able to match component with plugin broker result
     // when referenceContent is used
+    // TODO Consider reworking it according to Angel's proposal
+    // TODO it's needed to introduce source field into ChePlugin object with the following structure
+    // String registryUrl
+    // String id
+    // String reference
```

Comment:
My thought was to update `ChePlugin` to have a field `source` that is just a string with either
```json
  source: <reference>
```
or
```json
  source: id   // or registryUrl/id if applicable
```
but I agree that this makes sense as a future consideration; it would be easier to implement after the brokers refactor is mergd.

Reply:
> source: id   // or registryUrl/id if applicable

I believe having single string value would make a component to che-plugin matching harder...
Component has
```
id
registryUrl //optional - default will be used if omitted
reference
```

if plugin broker used default plugin registry - should it be included into source?
if yes - Che Server then should do the same defaultPluginRegistry+id, or cut id from source and then compare it.
But Che Server does not know for sure what is value of source field...
not sure if I clearly share my concerns BTW it's good that we agreed that it could be improved as a future consideration.

Reply:
Hmm, I'm not sure on the best approach. I'll need to think about it more -- my initial idea was to have `source` match the attribute on the `WorkspaceConfig`, so that it could just be parsed, but maybe that complicates things.

If adding the whole devfile into workspace config is a working solution, that may just be easier.